src/Pages/Shared/ConfigMapSecret/utils.ts

```diff
@@ -0,0 +1,661 @@
+                return CM_SECRET_STATE.INHERITED
+            default:
+                return configStage === ResourceConfigStage.Unpublished
+                    ? CM_SECRET_STATE.UNPUBLISHED
+                    : CM_SECRET_STATE.ENV
+        }
+    }
+
+    return configStage === ResourceConfigStage.Unpublished ? CM_SECRET_STATE.UNPUBLISHED : CM_SECRET_STATE.BASE
+}
+// HELPERS UTILS ----------------------------------------------------------------
+
+// FORM UTILS ----------------------------------------------------------------
+const secureValues = (data: Record<string, string>, decodeData: boolean, hideData: boolean): CMSecretYamlData[] => {
+    const decodedData = decodeData ? decode(data) : data
+    const hiddenData = hideData && Array(8).fill('*').join('')
+    return Object.keys(decodedData).map((k, id) => ({
+        k,
+        v:
+            typeof decodedData[k] === 'object'
+                ? hiddenData || YAMLStringify(decodedData[k])
+                : hiddenData || decodedData[k],
+        id,
+    }))
+}
+
+const processCurrentData = ({
+    configMapSecretData,
+    cmSecretStateLabel,
+    componentType,
+    isApprover,
+}: Pick<ConfigMapSecretFormProps, 'configMapSecretData' | 'cmSecretStateLabel' | 'componentType' | 'isApprover'>) => {
+    if (configMapSecretData.data) {
+        return secureValues(
+            configMapSecretData.data,
+            componentType === CMSecretComponentType.Secret && configMapSecretData.externalType === '',
+            !isApprover && componentType === CMSecretComponentType.Secret && configMapSecretData.unAuthorized,
+        )
+    }
+    if (cmSecretStateLabel === CM_SECRET_STATE.INHERITED && configMapSecretData?.defaultData) {
+        return secureValues(
+            configMapSecretData.defaultData,
+            componentType === CMSecretComponentType.Secret && configMapSecretData.externalType === '',
+            !isApprover && componentType === CMSecretComponentType.Secret && configMapSecretData.unAuthorized,
+        )
+    }
+
+    return CONFIG_MAP_SECRET_DEFAULT_CURRENT_DATA
+}
+
+const processExternalSubPathValues = ({
+    subPath,
+    externalType,
+    data,
+    esoSubPath,
+}: Pick<CMSecretConfigData, 'subPath' | 'externalType' | 'data' | 'esoSubPath'>) => {
+    if (subPath) {
+        if (data && externalType === CMSecretExternalType.KubernetesSecret) {
+            return Object.keys(data).join(', ')
+        }
+        if (esoSubPath) {
+            return esoSubPath.join(', ')
+        }
+    }
+    return ''
+}
+
+export const convertYAMLToKeyValuePair = (yaml: string): CMSecretYamlData[] => {
+    try {
+        const obj = yaml && YAML.parse(yaml)
+        if (typeof obj !== 'object') {
+            throw new Error()
+        }
+        const keyValueArray: CMSecretYamlData[] = Object.keys(obj).reduce((agg, k, id) => {
+            if (!k && !obj[k]) {
+                return CONFIG_MAP_SECRET_DEFAULT_CURRENT_DATA
```

Comment:
Can we do structuredClone before returing

Reply:
why?

Reply:
I believe methods are supposed to be pure, and we should not return object with their references, so if we change any key inside response this will change CONFIG_MAP_SECRET_DEFAULT_CURRENT_DATA as well

Reply:
we are not changing anything 
this returned data is being used as payload for api